### ISSUE TITLE
generic enums fail to produce correct schemas for variants

### ISSUE THREAD

**Issue by johnperry-math** [juhaku/utoipa#835]
## Background

1. Take the `todo-axum` crate
2. add a generic enum type with an alias, as recommended in docs:

           #[derive(ToSchema, Serialize, Deserialize, Clone)]
           #[aliases(TestEnumString = TestEnum<String>)]
           pub(super) enum TestEnum<T> {
               Video(T),
           }
3. add a field of this type to `struct Todo`:

               test: TestEnumString,

   ...and a corresponding element to the components:

               components(
                   schemas(todo::Todo, todo::TodoError, todo::TestEnumString)
               ),

4. run the example. swagger's OpenApi gives the following error:

       Resolver error at paths./todo.get.responses.200.content.application/json.schema.items.properties.test.oneOf.0.properties.Video.$ref
       Could not resolve reference: Could not resolve pointer: /components/schemas/T does not exist in document does not exist in document
5. 🤬

This makes sense, but...

## Question
How do I get utoipa to stop adding `T` to the OpenApi?

## Failed workaround
The docs _seem_ to say that enum variant configuration options for `#[schema(...)]` support...
> all variant specific configuration options e.g. if variant is UnnamedStruct then unnamed struct type configuration options are supported.

...but this does not seem to be true. Unnamed struct type configuration options include `as =`, so I tried `#[schema(as = Video(String))]` before `Video(T)`, and it failed with

    unexpected attribute: as, expected any of: example, default, title, format, value_type, rename, deprecated

## Current (working) workaround

1. Define yet another type,

           #[derive(ToSchema, Serialize, Deserialize, Clone)]
           pub(super) enum TestEnumString {
               Video(String),
           }
2. Replace `#[aliases(TestEnumString = TestEnum<String>)]` with `#[schema(as = TestEnumString)]`.
3. 🤩 

...but this seems a little hack-y.

I can include fuller source code if need be.

**Comment by RocketRace** [comment]
I stumbled into this issue myself as well.

In case it helps someone reading:

For my personal use case, I attempted to create an `enum NotRequired<T>` wrapper as a way to represent non-nullable, non-required optional fields, since `utoipa` automatically treated `Option<T>` fields as nullable even when setting `#[serde(default, skip_if="Option::is_none")]` attributes. I was able to work around this by instead using the `#[schema(schema_with)]` attribute like so:
```rs
#[derive(ToSchema)]
struct Example {
    #[schema(schema_with = String::schema)] // resulting schema has `field: string`
    field: Option<String>
}
```

**Comment by juhaku** [comment]
This will be fixed here with full generics support #1034
